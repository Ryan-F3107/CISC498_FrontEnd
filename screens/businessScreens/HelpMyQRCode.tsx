import React from 'react';
import { Text, View } from 'react-native';
import { IconButton } from 'react-native-paper';
import styles from '../../styling/styles';

function HelpMyQRCode({ navigation }) {
    return (
        <View style={styles.homeContainer}>

            {/*Close Button*/}
            <IconButton
                style={styles.closeButton}
                icon="close"
                size={35}
                color={'black'}
                onPress={() => { navigation.goBack() }}
            ></IconButton>

            {/*QR Code FAQ*/}
            <View style={styles.QRCodeContainer}>
                <Text style={styles.QRcodeTitle}> QR Code FAQ </Text>

                <Text style={styles.QRcodeQuestionTitle}> What happends when a customer scan a QR code? </Text>
                <Text style={styles.QRcodeAnswer}>
                    {'\n'}Scanning the QR code allows
                    {'\n'}customers to check into your business.
                    {'\n'}
                    {'\n'}The QR code sends information about
                    {'\n'} * your business
                    {'\n'} * customer information
                    {'\n'}
                </Text>

                <Text style={styles.QRcodeQuestionTitle}> How to display a QR code? </Text>
                <Text style={styles.QRcodeAnswer}>
                    {'\n'} 1.   Download the QR code by clicking
                    {'\n'}      the "Download PDF" button.
                    {'\n'}
                    {'\n'} 2.   Post it wherever your customers
                    {'\n'}      can see it well and scan it easily,
                    {'\n'}      such as at the entrance,
                    {'\n'}      at the cash register,
                    {'\n'}      and/or around a waiting area.

                    {'\n'}
                </Text>
            </View >
        </View >
    )
}
export default HelpMyQRCode;