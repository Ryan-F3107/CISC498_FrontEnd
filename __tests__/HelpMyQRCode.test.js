import React from 'react';
import renderer from 'react-test-renderer';
import HelpMyQRCode from '../screens/businessScreens/HelpMyQRCode';

describe("<HelpMyQRCode />", () => {
    it("has 2 children", () => {
        const tree = renderer.create(<HelpMyQRCode />).toJSON();
        expect(tree.children.length).toBe(2);
    });

    it('renders correctly', () => {
        renderer.create(<HelpMyQRCode />);
    });

    it('HelpMyQRCode testing with snapshot', () => {
        const tree = renderer.create(<HelpMyQRCode />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});