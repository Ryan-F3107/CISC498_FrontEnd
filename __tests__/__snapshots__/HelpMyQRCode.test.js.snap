// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HelpMyQRCode /> HelpMyQRCode testing with snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fafafa",
      "flex": 1,
      "paddingBottom": 20,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": undefined,
      }
    }
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "margin": 6,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 26.25,
            "height": 52.5,
            "width": 52.5,
          },
          undefined,
          Object {
            "alignSelf": "flex-end",
            "marginRight": 10,
            "marginTop": 40,
          },
        ],
      ]
    }
  >
    <View>
      <Text />
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fafafa",
        "flex": 1,
        "paddingHorizontal": 70,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "fontSize": 30,
          "paddingBottom": 10,
        }
      }
    >
       QR Code FAQ 
    </Text>
    <RCTScrollView
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      showsVerticalScrollIndicator={false}
    >
      <View>
        <Text
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
              "paddingBottom": 15,
              "paddingTop": 30,
            }
          }
        >
           What happens when a customer scan a QR code? 
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "#ebf2ff",
              "borderColor": "#a4cbde",
              "borderRadius": 10,
              "borderWidth": 1,
              "fontSize": 15,
              "padding": 10,
              "paddingBottom": 10,
            }
          }
        >
          

          Scanning the QR code allows
          

          customers to check into your business.
          

          

          The QR code sends information about
          

           * your business
          

           * customer's contact information and the total number of people who were checked in as one group
          

        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
              "paddingBottom": 15,
              "paddingTop": 30,
            }
          }
        >
           How to display a QR code? 
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "#ebf2ff",
              "borderColor": "#a4cbde",
              "borderRadius": 10,
              "borderWidth": 1,
              "fontSize": 15,
              "padding": 10,
              "paddingBottom": 10,
            }
          }
        >
          

           1.   Get a copy of your QR code by
          

                tapping the "Share PDF"
          

                button.
          

          

           2.   Post it wherever your customers
          

                can see it well and scan it easily,
          

                such as at the entrance,
          

                at the cash register,
          

                and/or around a waiting area.
          

        </Text>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
